core/src/cluster_info_repair_listener.rs

```diff
@@ -305,20 +305,30 @@ impl ClusterInfoRepairListener {
                         // a database iterator over the slots because by the time this node is
                         // sending the blobs in this slot for repair, we expect these slots
                         // to be full.
-                        if let Some(blob_data) = blocktree
-                            .get_data_blob_bytes(slot, blob_index as u64)
+
+                        if let Some(data_blob) = blocktree
+                            .get_data_blob(slot, blob_index as u64)
                             .expect("Failed to read data blob from blocktree")
                         {
-                            socket.send_to(&blob_data[..], repairee_tvu)?;
+                            socket.send_to(&data_blob.data[..], repairee_tvu)?;
                             total_data_blobs_sent += 1;
-                        }
 
-                        if let Some(coding_bytes) = blocktree
-                            .get_coding_blob_bytes(slot, blob_index as u64)
-                            .expect("Failed to read coding blob from blocktree")
-                        {
-                            socket.send_to(&coding_bytes[..], repairee_tvu)?;
-                            total_coding_blobs_sent += 1;
+                            if let Some(coding_header) = data_blob.get_coding_header() {
+                                for coding_blob_index in 0..coding_header.parity_count as u64 {
+                                    let coding_blob_opt = blocktree
+                                        .get_coding_blob(
+                                            slot,
+                                            coding_header.set_index,
+                                            coding_blob_index,
+                                        )
+                                        .expect("Failed to read coding blob from blocktree");
+
+                                    if let Some(coding_blob) = coding_blob_opt {
```

Comment:
if there are tests for this code, it needs to be updated to expect these coding blobs

Reply:
there isn't anything that tests for the particular blobs sent.